src/compiler/checker.ts

```diff
@@ -4973,6 +5031,7 @@ namespace ts {
                 if (relation === assignableRelation) {
                     if (isTypeAny(source)) return Ternary.True;
                     if (source === numberType && target.flags & TypeFlags.Enum) return Ternary.True;
+                    if (source.flags & TypeFlags.Boolean && target.flags & TypeFlags.Boolean && (source.flags & TypeFlags.PredicateType || target.flags & TypeFlags.PredicateType)) return Ternary.True;

```

Comment:
Oh, so after adding the value types, if you have a set containing the types `boolean`, `true`, `false`, and type predicates, which pairs would you want to exclude from the assignability relation?